File: server/src/main/java/com/fptmarket/service/OrderService.java
package com.fptmarket.service;

import com.fptmarket.dto.request.OrderRequest;
import com.fptmarket.dto.response.OrderResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface OrderService {
    OrderResponse createOrder(OrderRequest request);
    Page<OrderResponse> getMyOrders(Pageable pageable);
    OrderResponse getOrderById(Long id);
    OrderResponse cancelOrder(Long id);
    Page<OrderResponse> getAllOrders(Pageable pageable);
}

File: client/src/services/orderService.js
import api from './api';

const orderService = {
  createOrder: async (orderData) => {
    const response = await api.post('/api/orders', orderData);
    return response.data;
  },

  getMyOrders: async (page = 0, size = 10) => {
    const response = await api.get(`/api/orders/my?page=${page}&size=${size}`);
    return response.data;
  },

  getOrderById: async (id) => {
    const response = await api.get(`/api/orders/${id}`);
    return response.data;
  },

  cancelOrder: async (id) => {
    const response = await api.put(`/api/orders/${id}/cancel`);
    return response.data;
  }
};

export default orderService;
